universal-application-tool-0.0.1/test/services/program/ProgramServiceImplTest.java

```diff
@@ -0,0 +1,136 @@
+package services.program;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+import java.util.Locale;
+import java.util.Optional;
+import org.junit.Before;
+import org.junit.Test;
+import repository.WithResettingPostgresContainer;
+import services.question.QuestionDefinition;
+
+public class ProgramServiceImplTest extends WithResettingPostgresContainer {
```

Comment:
Can we add a test where the block with given ID does not exist on the program?

Reply:
Done

Reply:
Done